grafana/grafana
# DataSourceWithBackend: Add `queryGroupId` to find correlated/related queries

**What is this feature?**

We are currently adding query chunking to the Loki datasource. This means a 30day query is split into 30 individual 1day queries. This PR is adding a `correlationId`, which is send via a HTTP header, to still see the relations/correlations of those individual queries.

**Why do we need this feature?**

Queries might potentially influence each other. This ID is good to detect relevant queries.

**Special notes for your reviewer**:

This implementation is quite generic, as query chunking might be something also other datasource can implement in the future.

**How to test this:**

1. Setup a Loki datasource
2. `make devenv sources=loki`
3. Go to Explore
4. Run the following query over the last 7 days: `sum(count_over_time({job="\"grafana/data\""} |= `` [$__interval]))`
5. Observe network requests in e.g. the network tab of devtools
6. See 7 1day queries in the network tab. Each of the HTTP request has the same `X-Correlation-Id` header.
